src/stories/atoms/Shape/Shape.stories.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import { Shape } from 'stories/atoms/Shape/Shape';
+
+export default {
+  title: 'Design System/Atoms/Shape',
+  component: Shape,
+} as ComponentMeta<typeof Shape>;
+
+const Template: ComponentStory<typeof Shape> = (args) => <Shape {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
```

Comment:
And this is the other part of my comment above. `Default.args` are the unique inputs for this instance of `Template`. Again, we see the `Storybook` syntax taking advantage of an obscure feature of JavaScript (`hoisting`) to create something elegant and streamlined.

So we've already declared `Default` which is accepting a single `args` argument that we initially declare as just an empty object:

```ts
export const Default = Template.bind({});
```
We then use `hoisting` to re-declare the value of that empty `args` object. `Hoisting` allows us to use a variable before its been declared. Essentially we've called this:

```ts
export const Default = Tempalte.bind({
 backgroundColor: '#c4c4c3",
 shapeHeight: '250',
 shapeWidth: '250',
})
```
And what this allows us to ultimately do is have a `Default` version of our arguments that we can then spread (`...`) into other `stories`.

Reply:
It was great taking a dive into this. Basically, we first create the `Template` function. We then create a copy of that `Template` function using `.bind()`. `.bind({})` returns a copy of the function with an empty object as the argument to the `args` parameter. We assign this copied function to the variable `Default` and appear to export it, before we assign it our desired 'props'. However, due to hoisting, we can later assign our `Default.args` parameter to be the `props` we wish to see in our rendered `story`. 

Where I run into some confusion is `hoisting`. Here is how I understand it working in our case. `Hoisting` allows us to reference a variable before it is created (variable hoisting). When we declare, initialize and export `Default` we use the `args` parameter and set it to an empty object-- since `export const Default = Template.bind({});` is essentially equal to `export const Default = ({ }) => <Shape { } />`. So we appear to export our story before it has received its desired `props` as `args`. We then declare `Default.args` whose declaration is hoisted to the global scope, this is what allows us to reference our `args` variable during our export before our desired values are assigned . Once the operation reaches the `Default.arg` declaration, the `{...arg}` values are initialized and passed to the `Default` function, the function with these new arguments is then exported. This function is then processed by `Storybook` to generate our named story.

Ultimately, who chose this architecture because we wanted to create a `Default.args` object that holds our default parameters that we could then spread into our other named exports.  

Phew! I feel like I may be missing something. `Hoisting` is definitely a concept I will spend more time with!